scio-test/src/it/scala/com/spotify/scio/benchmarks/ScioBenchmark.scala

```diff
@@ -267,112 +270,128 @@ class DatastoreLogger(metricsToCompare: Set[String]) extends BenchmarkLogger[Try
-                )
-                .build())
-          }
-          .map(_.getBatch.getEntityResults(0).getEntity)
-      case None =>
-        val comparisonMetrics = Storage.runQuery(
+  def latestBenchmark(benchmarkName: String): Option[Entity] =
+    benchmarksByBuildNum(benchmarkName, Nil).headOption
+
+  def benchmarksByBuildNum(benchmarkName: String, buildNums: List[Long]): List[Entity] = {
+    val results = buildNums match {
+      case Nil =>
+        //fetch latest
+        Storage.runQuery(
           RunQueryRequest
             .newBuilder()
             .setGqlQuery(
               GqlQuery
                 .newBuilder()
                 .setAllowLiterals(true)
-                .setQueryString(OrderByBuildNumQuery.format(benchmarkName))
+                .setQueryString(latestBuildNumQuery(benchmarkName))
                 .build()
             )
-            .build())
+            .build()) :: Nil
+
+      case bns =>
+        bns.map { bn =>
+          Storage.runQuery(
+            RunQueryRequest
+              .newBuilder()
+              .setGqlQuery(
+                GqlQuery
+                  .newBuilder()
+                  .setAllowLiterals(true)
+                  .setQueryString(buildNumQuery(benchmarkName, bn))
+                  .build()
+              )
+              .build())
+        }
+    }
+
+    results
+      .flatMap(_.getBatch.getEntityResultsList.asScala)
+      .sortBy(_.getEntity.getKey.getPath(0).getName)
+      .map(_.getEntity)
+  }
 
-        comparisonMetrics.getBatch.getEntityResultsList.asScala
-          .sortBy(_.getEntity.getKey.getPath(0).getName)
-          .map(_.getEntity)
+  def printMetricsComparison(previous: Entity, current: Entity): Unit = {
+    val buildNum: Entity => String = _.getKey.getPath(0).getName
+    val entityPropValue: Entity => String => Double = e =>
+      k => e.getPropertiesMap.get(k).getStringValue.toDouble
+
+    val opName = current.getKey.getPath(0).getKind.substring(Kind.length + 1)
+    PrettyPrint.printSeparator()
+    PrettyPrint.print("Benchmark", opName)
+    PrettyPrint.print("BuildNum",
+                      "%15s%15s%15s".format(buildNum(previous), buildNum(current), "Delta"))
+
+    val previousProps = entityPropValue(previous)
+    val currentProps = entityPropValue(current)
+    metricsToCompare.foreach { k: String =>
+      val prev = previousProps(k)
+      val curr = currentProps(k)
+      val delta = (curr - prev) / prev * 100.0
+      val signed = if (delta.isNaN) {
+        "0.00%"
+      } else {
+        (if (delta > 0) "+" else "") + "%.2f%%".format(delta)
+      }
+      PrettyPrint.print(k, "%15.2f%15.2f%15s".format(prev, curr, signed))
     }
+  }
 
   // TODO: move this to email generator
-  def printMetricsComparison(benchmarks: Iterable[String],
-                             buildNums: Option[(String, String)] = None): Unit = {
+  def printMetricsComparison(benchmarks: Iterable[String], buildNums: List[Long]): Unit =
```

Comment:
I feel like these two `def printMetricsComparison` methods should have clearer function names to differentiate them? since this method is about comparing Benchmarks, whereas the function on line 354 is directly comparing two datastore Entities. 

also on that note, if we're comparing benchmarks across multiple builds, I think my assumption would be that the delta percentages are all relative to the oldest build. I might be alone in that though :P

Reply:
the idea is to in a follow up PR not expose the Entity ... coz the way it is right now with the usage of `BenchmarkResult` is just messy. So:

```
def printMetricsComparison(prev: Entity, curr: Entity)

becomes

def printMetricsComparison(prev: BenchmarkResult, curr: BenchmarkResult)
```

well, i actually like the idea of seeing the deltas between consecutive builds and this doesn't invalidate the idea of adding the deltas relative to the older one.

Reply:
oh, cool, didn't know there would be a followup PR. makes sense! excited to see the benchmark code cleaned up a bit 💯